### ISSUE TITLE
Mac OS/X Yosemite issues

### ISSUE THREAD

**Issue by diekhans** [ga4gh/compliance#33]
Installing JVMs is still not a solved problem.  We should probably have a JVM install hints section.

For Mac OS/X, Apple no longer supports java and 1.8 needs to be obtain from Orcale.

I install Maven via MacPorts, but we should also include instructions for Brew.

Maven doesn't pick up the right version of java classes due to JAVA_HOME not being set.
Setting environment variables on OS/X has been rather complex and  setting in the .bashrc
with this command is what is suggested:

        export JAVA_HOME=$(/usr/libexec/java_home)

So that wasted an hour and got me to the point of running mvn

**Comment by hjellinek** [comment]
Yeah, thank sjobs@apple.com for that first part.  Things were easier when Apple and Oracle got along.

Somewhere I did write directions for Brew... it's in `UpdatingTheSchemas.md`.  Obviously that needs to be copied to the quickstart guide that @macieksmuga is working on. 

I don't have `JAVA_HOME` or any other Java-related env var set - I find that's error-prone, plus dealing with `.bash*` files is a hassle.  It sounds like you've gotten past that, though.  Maybe the `Brew` version of Maven knows where to look?  Here's what mine says:

    $ mvn --version
    Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T04:57:37-07:00)
    Maven home: /usr/local/Cellar/maven/3.3.3/libexec
    Java version: 1.8.0_51, vendor: Oracle Corporation
    Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_51.jdk/Contents/Home/jre
    Default locale: en_US, platform encoding: UTF-8
    OS name: "mac os x", version: "10.10.4", arch: "x86_64", family: "mac"
    $ which mvn
    /usr/local/bin/mvn

**Comment by hjellinek** [comment]
More info for reference/quickstart:

Download the Java SE Development Kit (JDK), currently found [here](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html).  Get the latest JDK 8 version available.  Run the installer.

(Make sure not to confuse this with the Java SE Runtime Environment (JRE), which is a subset of the JDK.)

**Comment by diekhans** [comment]
Herb Jellinek <notifications@github.com> writes:
> I don't have JAVA_HOME or any other Java-related env var set - I find that's
> error-prone, plus dealing with .bash* files is a hassle. It sounds like you've
> gotten past that, though. Maybe the Brew version of Maven knows where to look?

Dealing with .bashrc is easy, mine works on multiple OSes, at
multiple institutions, in multiple countries (unlike *plist).

JAVA* env is a nightmare, but that is what the internet says to
do

> Here's what mine says:
> 
> $ mvn --version
> Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T04:57:37-07:00)
> Maven home: /usr/local/Cellar/maven/3.3.3/libexec
> Java version: 1.8.0_51, vendor: Oracle Corporation
> Java home: /Library/Java/JavaVirtualMachines/jdk1.8.0_51.jdk/Contents/Home/jre

what happens if you do: echo $JAVA_HOME ?

mvn is actually a bash script, is it set in there?

**Comment by hannes-ucsc** [comment]
`/usr/libexec/java_home` is the right approach. I use this wrapper around it in my `~/.profile`:

```
# Java
#
function jdk() {
    if [ $# -ne 0 ]; then
        removeFromPath '/System/Library/Frameworks/JavaVM.framework/Home/bin'
        if [ -n "${JAVA_HOME+x}" ]; then
            removeFromPath $JAVA_HOME
        fi
        export JAVA_HOME=`/usr/libexec/java_home -v $@`
        export PATH=$JAVA_HOME/bin:$PATH
        java -version
    else
        /usr/libexec/java_home -V
    fi
}

function removeFromPath() {
    export PATH=$(echo $PATH | sed -E -e "s;:$1;;" -e "s;$1:?;;")
}

jdk 1.7
```

That allows me to switch between JDK's quickly within one shell session. Running just `jdk` will list installed JDKs. Passing a JDK name will switch to that JDK. As you can see, I use 1.7 by default.

If you need to set `JAVA_HOME` outside of shell sessions, e.g. for an IDE started via Finder, you could use my EnvPane.

**Comment by hjellinek** [comment]
I hate that when I edit `.bashrc` I need to source it or restart my shell.  (I know there are shells that solve this problem, but I don't want to switch from bash.)

As mentioned above, my env vars are clean:

    $ echo $JAVA_HOME

    $ 

But you guess correctly about `mvn`:

    $ grep -i java `which mvn`
    JAVA_HOME="${JAVA_HOME:-$(/usr/libexec/java_home)}" exec "/usr/local/Cellar/maven/3.3.3/libexec/bin/mvn" "$@"

That "`cellar`" stuff is Brew-related.  Seems like Brew might be a better solution than MacPorts?  I've had similar experiences with other tools and finally switched entirely to Brew.

**Comment by hannes-ucsc** [comment]
Ditch MacPorts, adopt Brew. It uses git instead of svn, installs to '/usr/local' rather than '/opt/local', rarely needs root, has more pre-built binaries (aka bottles), makes it much easier to customize a build (aka 'formula'), and is not dying.

**Comment by diekhans** [comment]
Herb Jellinek <notifications@github.com> writes:
> That "cellar" stuff is Brew-related. Seems like Brew might be a better solution
> than MacPorts? I've had similar experiences with other tools and finally
> switched entirely to Brew.

Last I looked, Brew had a small fraction of the number of
packages that MacPorts had, but Brew seems to do better job with
java.

**Comment by hannes-ucsc** [comment]
You can't just look at the master repo. Formulas that duplicate functionality already shipped with OS X is in a different repo (aka "dupes") and so is stuff with multiple versions (aka "versions"). These repos need to be tapped separately. Check out braumeister.org. Also, what good are 24k packages if many of them are unmaintained? That's what I often found with MacPorts.

**Comment by david4096** [comment]
I used Oracle's JDK and `brew install maven`. :)
